java/runtime/src/main/java/org/ray/runtime/Worker.java

```diff
@@ -95,7 +96,15 @@ public void execute(TaskSpec spec) {
         currentActorId = returnId;
       }
     } finally {
-      runtime.getWorkerContext().setCurrentTask(null, null);
+      if (spec.isActorTask() || spec.isActorCreationTask()) {
+        // Don't need to reset current driver id if the worker is an actor.
+        // Because the following tasks should all have the same driver id.
+        runtime.getWorkerContext().setCurrentTask(null, spec.driverId, null);
+      } else {
+        // For normal task, we should set current task id to null as well.
+        runtime.getWorkerContext().setCurrentTask(null, null, null);
+      }
+
       Thread.currentThread().setContextClassLoader(oldLoader);
```

Comment:
Sorry I'm not clear on what `classLoader` is, but it seems like you could get rid of this line and pass in `oldLoader` in the `setCurrentTask` lines above?

Reply:
@stephanie-wang `classLoader` is used to tell JVM where and how to load a class into memory. We use different class loaders for different drivers. So different drivers can have classes with the same name without conflicts. 
@jovany-wang `classLoader` should be set and reset in the same way as the driver id.